src/app/LayoutCommon.astro

```diff
@@ -8,9 +8,25 @@ const { class: clazz, metaData } = Astro.props;
 
 const initData: any = {};
 
-const envKeys = [...Object.keys(process.env), ...Object.keys(import.meta.env)];
-envKeys.forEach((item) => {
-	initData[item] = process.env[item] || import.meta.env[item];
+// Step 1: List of keys that should NOT be exposed to the frontend
+const excludeKeys = ['PUBLIC_CRYPTO_PRIVATE_KEY', 'PUBLIC_KEYCLOAK_MANAGEMENT_CLIENT_ID', 'PUBLIC_KEYCLOAK_MANAGEMENT_CLIENT_SECRET', 'PUBLIC_ALLOW_DOMAIN'];
+
+// Step 2: Get all environment keys
+const allEnvKeys = [...Object.keys(process.env), ...Object.keys(import.meta.env)];
+
+// Step 3: Separate 'exposed' keys and 'excluded' keys
+const exposedEnvKeys = allEnvKeys.filter((key) => !excludeKeys.includes(key));
```

Comment:
merge these two filter conditions.

Reply:
As we have already created an array for excludeEnvKeys, there is no need to add another array for excluding environment variables. Therefore, there will be only one condition.